core/staking_verifier.go

```diff
@@ -264,7 +266,13 @@ func VerifyAndUndelegateFromMsg(
 			if err := wrapper.SanityCheck(
 				staking.DoNotEnforceMaxBLS,
 			); err != nil {
-				return nil, err
+				// allow self delegation to go below min self delegation
+				// but set the status to inactive
+				if strings.Contains(err.Error(), staking.ErrInvalidSelfDelegation.Error()) {
```

Comment:
Instead of string compare, breaking the abstraction of error message, just check if the error value is == ErrInvalidSelfDelegation

Reply:
Or also, check out  https://golang.org/pkg/errors/#Unwrap If there are cases where we want to keep wrapping because of the extra debug info